gen/enum.go

```diff
@@ -265,6 +275,34 @@ func enumItemName(enumName string, spec *compile.EnumItem) (string, error) {
 	return enumName + name, err
 }
 
+// enumItemLabelName returns the actual name used for serialization/deserialization
+// default to EnumItem.Name, override by the value of GoLabel
+func enumItemLabelName(spec *compile.EnumItem) string {
+	labelName := spec.Name
+	val, ok := spec.Annotations[GoLabel]
+	if ok && len(val) > 0 {
+		labelName = val
+	}
+	return labelName
+}
+
+// validateEnumUniqueNames apply name label GoLabel and raise error if there's
+// duplicates in resolved enum item names
+func validateEnumUniqueNames(spec *compile.EnumSpec) error {
+	items := spec.Items
+	used := make(map[string]bool, len(items))
```

Comment:
When using a `map` as a set, we prefer to use `struct{}` instead of bool so
that you don't accidentally rely on the value of the boolean. Plus `struct{}`
has zero runtime cost.

```
used := make(map[string]struct{}, len(items))
...
used[itemName] = struct{}{}
```

Reply:
thanks! Pretty neat trick, and good to know about the zero runtime cost part :)
```go
func BenchmarkBool(b *testing.B) {
	mp := make(map[string]bool)
	for i := 0; i < b.N; i++ {
		if _, ok := mp[string(i)]; ok {
			mp[string(i)] = true
		}
	}
}

func BenchmarkStruct(b *testing.B) {
	mp := make(map[string]struct{})
	for i := 0; i < b.N; i++ {
		if _, ok := mp[string(i)]; ok {
			mp[string(i)] = struct{}{}
		}
	}
}
```
```
▶ go test -bench=Benchmark -count=1 -benchmem bench_test.go
goos: darwin
goarch: amd64
BenchmarkBool-8     	200000000	         7.00 ns/op	       0 B/op	       0 allocs/op
BenchmarkStruct-8   	200000000	         7.31 ns/op	       0 B/op	       0 allocs/op
PASS
```